src/MetadataReader/AnnotationAttributeDualReader.php

```diff
@@ -0,0 +1,121 @@
+<?php declare(strict_types = 1);
+
+namespace Apitte\Core\MetadataReader;
+
+use Doctrine\Common\Annotations\AnnotationReader;
+use Doctrine\Common\Annotations\Reader;
+use Koriym\Attributes\AttributeReader;
+use ReflectionClass;
+use ReflectionMethod;
+use ReflectionProperty;
+
+/**
+ * This reader reads BOTH annotations and attributes above one elements at the same time,
+ * and merge them together.
+ * Compared to @see DualReader that only prefers one or the other.
+ */
+final class AnnotationAttributeDualReader implements Reader
```

Comment:
@koriym Hi, I came to a bug when only annotations or attributes are used. The `DualReader` can handle only one of them, depending on PHP version.

This caused missing annotations on PHP 8. I think more reliable solution would be class like this one - that reads both annotations and attributes on PHP 8, if available.  That's what I expected from `DualReader` class and that's where it failed my expectations. But maybe there is a better way to do it.

What do you think?

Reply:
As you pointed out! It's a bug and needs to be fixed.

Reply:
Yes, exactly. It fails for the arrays

Reply:
Issue created. https://github.com/koriym/Koriym.Attributes/issues/10